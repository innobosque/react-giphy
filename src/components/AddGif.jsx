import { useState } from 'react';
import PropTypes from 'prop-types';
/**
 * 
 * @param {function} Función de comunicación (emit) donde se pasa la información del control input al padre 
 * @returns Formulario con el control de tipo input
 */
export const AddGif = ({ onNewGif }) => {

    const [value, setValue] = useState('');

    const handleSubmit = (event) => {
        event.preventDefault();
        const newValue = value.trim();
        if (newValue.length < 1) return;
        onNewGif(newValue);
        setValue('');
    }

    const handleChange = ({ target }) => {
        setValue(target.value);
    }

    return (
        <form onSubmit={handleSubmit}>
            <input type="search"
                onChange={handleChange}
                placeholder="🍳"
                value={value} />
        </form>
    );
}

AddGif.propTypes = {
    value: PropTypes.string
}