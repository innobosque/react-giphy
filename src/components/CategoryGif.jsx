import { useFetchGifs } from "../hooks/useFetchGifs";
export const CategoryGif = ({ category }) => {
    const {isLoading,images} = useFetchGifs(category);
    return (
        <>
            {isLoading && 'Cargando'}
            {images.map(image=>(
                <img className="w-full h-full object-cover" key={image.id} src={image.url} alt={image.title} />
            ))}
        </>
    )
}