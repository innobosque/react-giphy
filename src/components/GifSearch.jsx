import { useState } from 'react';
import PropTypes from 'prop-types';
/**
 * 
 * @param {function} Función de comunicación (emit) donde se pasa la información del control input al padre 
 * @returns Formulario con el control de tipo input
 */
export const GifSearch = ({ onNewGif }) => {

    const [value, setValue] = useState('');

    const handleSubmit = (event) => {
        event.preventDefault();
        const newValue = value.trim();

        if (!!!newValue.length) return setValue('');

        onNewGif(newValue);
        setValue('');
    }

    const handleChange = ({ target }) => {
        setValue(target.value);
    }

    return (
        <form onSubmit={handleSubmit}>
            <input className='w-full mt-8 py-1 px-2 outline-none bg-white rounded-md placeholder-black-400' type="search"
                onChange={handleChange}
                placeholder="Buscar GIF"
                value={value} />   
        </form>
    );
}

GifSearch.propTypes = {
    value: PropTypes.string
}