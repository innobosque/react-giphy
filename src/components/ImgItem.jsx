import Proptypes from 'prop-types';
export const ImgItem = ({url,title}) => (
    <img className="w-full h-full object-cover" src={url} alt={title} />
);

ImgItem.propTypes = {
    url: Proptypes.string.isRequired,
    title: Proptypes.string.isRequired,
}

ImgItem.defaultProps = {
    title: ''
}
