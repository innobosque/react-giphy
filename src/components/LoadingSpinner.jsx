import PropTypes from 'prop-types';
export const LoadingSpinner = ({widthHeight}) => {
  return (   
    <svg className="svg-spinner block m-auto" xmlns="http://www.w3.org/2000/svg" width={widthHeight} height={widthHeight} viewBox="0 0 48 48">
    <circle cx="24" cy="4" r="4" fill="#fff"/>
    <circle cx="12.19" cy="7.86" r="3.7" fill="#0ffbf2"/>
    <circle cx="5.02" cy="17.68" r="3.4" fill="#0ef7e4"/>
    <circle cx="5.02" cy="30.32" r="3.1" fill="#0ef3d7"/>
    <circle cx="12.19" cy="40.14" r="2.8" fill="#0eefc9"/>
    <circle cx="24" cy="44" r="2.5" fill="#0eebbc"/>
    <circle cx="35.81" cy="40.14" r="2.2" fill="#0de7af"/>
    <circle cx="42.98" cy="30.32" r="1.9" fill="#0de3a1"/>
    <circle cx="42.98" cy="17.68" r="1.6" fill="#0ddf94"/>
    <circle cx="35.81" cy="7.86" r="1.3" fill="#0cdb86"/>
  </svg>
  )
}

LoadingSpinner.propTypes = {
    widthHeight: PropTypes.string 
}

LoadingSpinner.defaultProps = {
    widthHeight: "24" 
}