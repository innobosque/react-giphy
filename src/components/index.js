export * from './GifSearch';
export * from './GifEndpoint/GifMapSearch';
export * from './GifEndpoint/GifSearchedItem';
export * from './LoadingSpinner';