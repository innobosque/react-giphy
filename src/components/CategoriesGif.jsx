import { CategoryGif } from "./CategoryGif";

export const CategoriesGif = ({ categories }) => {

    return (
        <section>
            <h2 className="text-center text-2xl font-bold">Categor√≠as</h2>
            {
                categories.map(category => (
                    <div className="grid grid-cols-5 gap-5" key={category} >
                        <h3 className="col-span-5 p-5 uppercase">{category}</h3>
                        <CategoryGif category={category} />
                    </div>
                ))
            }

        </section>
    )
}