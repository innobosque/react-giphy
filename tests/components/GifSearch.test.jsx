import { fireEvent, render, screen } from "@testing-library/react";
import { GifSearch } from "../../src/components/GifSearch";

describe('Componente GitImage',()=>{
    test('Hacer un match a un snapshot',()=>{
        const {container} = render(<GifSearch onNewGif={()=>{}} />);
        expect(container).toMatchSnapshot();
    });

    test('Cambiar el valor de la caja de texto',()=>{
        render(<GifSearch onNewGif={()=>{}} />);
        const input = screen.getByRole('searchbox');
        //disparemos un evento
        fireEvent.input(
            input,
            {
                target: {value:'Xurxo'}
            });
        expect(input.value).toBe('Xurxo');
            
        //screen.debug();
    });
});