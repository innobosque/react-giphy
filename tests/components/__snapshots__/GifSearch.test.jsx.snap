// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente GitImage Hacer un match a un snapshot 1`] = `
<div>
  <form>
    <input
      class="w-full mt-8 py-1 px-2 outline-none bg-white rounded-md placeholder-black-400"
      placeholder="Buscar GIF"
      type="search"
      value=""
    />
  </form>
</div>
`;
